File: src/main/java/architecture/goldenboughs_lib/client/model/item/GeoItemModel.java
package architecture.goldenboughs_lib.client.model.item;

import architecture.goldenboughs_lib.client.model.GeoModelExpand;
import net.minecraft.resources.ResourceLocation;
import software.bernie.geckolib.animatable.GeoAnimatable;

public class GeoItemModel<T extends GeoAnimatable> extends GeoModelExpand<T> {
	public GeoItemModel(ResourceLocation pathName) {
		this(pathName, pathName, pathName);
	}

	public GeoItemModel(ResourceLocation modelPath, ResourceLocation textureName, ResourceLocation animationsName) {
		super(model(getPath(modelPath)), texturePath(getPath(textureName)), animationsPath(getPath(animationsName)));
	}

	public static ResourceLocation getPath(ResourceLocation modelPath) {
		return modelPath.withPrefix("item/");
	}
}

File: src/main/kotlin/architecture/goldenboughs_lib/client/model/item/GeoItemModel.kt
package architecture.goldenboughs_lib.client.model.item

import architecture.goldenboughs_lib.client.model.GeoModelExpand
import net.minecraft.resources.ResourceLocation
import software.bernie.geckolib.animatable.GeoAnimatable

class GeoItemModel<T : GeoAnimatable>(
	modelPath: ResourceLocation,
	texturePath: ResourceLocation,
	animationsPath: ResourceLocation
) : GeoModelExpand<T>(
	modelPath(getPath(modelPath)),
	texturePath(getPath(texturePath)),
	animationsPath(getPath(animationsPath))
) {
	constructor(pathName: ResourceLocation) : this(pathName, pathName, pathName)

	companion object {
		@JvmStatic
		fun getPath(modelPath: ResourceLocation): ResourceLocation {
			return modelPath.withPrefix("item/")
		}
	}
}
